sentry_sdk/integrations/openai.py

```diff
@@ -0,0 +1,228 @@
+from __future__ import absolute_import
+
+from sentry_sdk._types import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, Iterable, List, Optional, Callable, Iterator
+    from sentry_sdk.tracing import Span
+
+from sentry_sdk._functools import wraps
+from sentry_sdk.hub import Hub
+from sentry_sdk.integrations import DidNotEnable, Integration
+from sentry_sdk.utils import logger, capture_internal_exceptions
+
+try:
+    from openai.resources.chat.completions import Completions
+    from openai.resources import Embeddings
+
+    if TYPE_CHECKING:
+        from openai.types.chat import ChatCompletionMessageParam, ChatCompletionChunk
+except ImportError:
+    raise DidNotEnable("OpenAI not installed")
+
+try:
+    import tiktoken  # type: ignore
+
+    enc = tiktoken.get_encoding("cl100k_base")
+
+    def count_tokens(s):
+        # type: (str) -> int
+        return len(enc.encode_ordinary(s))
+
+    logger.debug("[OpenAI] using tiktoken to count tokens")
+except ImportError:
+    logger.info(
+        "The Sentry Python SDK requires 'tiktoken' in order to measure token usage from some OpenAI APIs"
+        "Please install 'tiktoken' if you aren't receiving token usage in Sentry."
+        "See https://docs.sentry.io/platforms/python/guides/openai/ for more information."
+    )
+
+    def count_tokens(s):
+        # type: (str) -> int
+        return 0
+
+
+COMPLETION_TOKENS = "completion_tоkens"
+PROMPT_TOKENS = "prompt_tоkens"
+TOTAL_TOKENS = "total_tоkens"
```

Comment:
Most of our span data fields are prefixed with something for easier identification. We should do this here to. 
Existing span data fields are documented here: https://develop.sentry.dev/sdk/performance/span-data-conventions/

(we try to align as closely to OpenTelemetry's semantic conventions for attributes (link in the docs page I linked) as possible, so maybe there is already something in there. so please check those)